submissions/Oleksii_Drohachov/Memory_pair_game/script.js

```diff
@@ -0,0 +1,203 @@
+const mainContainer = document.querySelector(".container");
+const modal = document.querySelector(".modal__container");
+const gameboard = document.querySelector(".gameboard");
+const scoreCount = document.querySelector(".score__container");
+
+// creating modal window
+const modalWrapper = document.createElement("div");
+modalWrapper.classList.add("modal__wrapper");
+const modalTitle = document.createElement("p");
+modalTitle.classList.add("modal__title");
+
+const modalBtn = document.createElement("button");
+modalBtn.classList.add("modal__btn");
+
+modalBtn.addEventListener("click", startGame);
+const closeBtn = document.createElement("button");
+closeBtn.classList.add("modal__btn");
+closeBtn.textContent = "NO, THANKS";
+closeBtn.addEventListener("click", hideModal);
+
+modalWrapper.append(modalTitle);
+modalWrapper.append(modalBtn);
+modalWrapper.append(closeBtn);
+
+modal.append(modalWrapper);
+
+renderModal(
+  "You are welcome, my friend ! Wanna play some memory game to endure your mind and become more powerfull ? Try to find pairs of cards as fast as you can! Tap the button to begin... ",
+  "START THE GAME"
+);
+
+function renderModal(titleText, btnText) {
+  modalTitle.textContent = titleText;
+  modalBtn.textContent = btnText;
+  modal.classList.add("shown");
+}
+
+function hideModal() {
+  modal.classList.remove("shown");
+}
+
+function startGame() {
+  scoreCount.innerHTML = "";
+  gameboard.innerHTML = "";
+  hideModal();
+  prepareGame();
+}
+
+function prepareGame() {
```

Comment:
Your function calls `prepareGame` but inside this function, you describe all game logic.
Single responsibility - your functions should do only one job. As an example function in which you fetch users should only fetch them and not render, transform or process them in other ways. All these things should be done in another place, outside your function. The same applied to the sort function, which usually does all types of sorting 😉